components/script/dom/performance.rs

```diff
@@ -273,10 +287,70 @@ impl Performance {
             o.notify();
         }
     }
-
     fn now(&self) -> f64 {
         (time::precise_time_ns() - self.navigation_start_precise).to_ms()
     }
+    fn can_add_resource_timing_entry(&self) -> bool {
+        self.resource_timing_buffer_current_size.get() <=
+            self.resource_timing_buffer_size_limit.get()
+    }
+    fn copy_secondary_resource_timing_buffer(&self) {
+        while self.can_add_resource_timing_entry() && !self
+            .resource_timing_secondary_entries
+            .borrow_mut()
+            .is_empty()
+        {
+            self.queue_entry(
+                &(*(self
+                    .resource_timing_secondary_entries
+                    .borrow_mut()
+                    .pop_front()
+                    .unwrap())),
+                true,
+            );
+        }
+    }
+    fn fire_buffer_full_event(&self) {
+        while !self
+            .resource_timing_secondary_entries
+            .borrow_mut()
+            .is_empty()
```

Comment:
Same as before, there is no change to the collection here, so we don't need to mutably borrow it.